src/main/java/com/jcraft/jsch/UserAuthPassword.java

```diff
@@ -162,6 +162,13 @@ public boolean start(Session session) throws Exception {
               throw new JSchPartialAuthException(Util.byte2str(foo));
             }
             session.auth_failures++;
+
+            // Do not ask re-entering password after the first failure. Useful when authenticating
+            // with a script that uses the same password every time.
+            if (Boolean.parseBoolean(JSch.getConfig("user_auth_single_attempt"))) {
```

Comment:
This needs to use `session.getConfig()`, not `JSch.getConfig()` so that it can be set on a per Session basis.

I believe you also need to add the config item to `Session.applyConfig()` via a `Session.checkConfig()` call.
See https://github.com/mwiede/jsch/blob/d064f44b5ae4c9447908826df8f5f5eba7ee5a84/src/main/java/com/jcraft/jsch/Session.java#L3207.